forge-game/src/main/java/forge/game/player/Player.java

```diff
@@ -629,6 +629,7 @@ public final boolean payLife(final int lifePayment, final SpellAbility cause, fi
         }
 
         loseLife(lifePayment, false, false);
+        cause.setPaidLife(lifePayment);
```

Comment:
i don't know if this is the right part for this?

shouldn't that only be set for when life is paid as cost?

Reply:
gonna commit this, just update this/open an issue later on if needed be.